src/MSBuild/PerformanceLogEventListener.cs

```diff
@@ -49,16 +50,28 @@ internal static PerformanceLogEventListener Create()
                 // we if we do this work in the constructor, and don't get an opportunity to call Dispose, the
                 // EventSources will remain enabled even if there aren't any consuming EventListeners.
 
+                Console.WriteLine("!!! Creating PerformanceLogEventListener");
+
                 // Check to see if we should enable the event listener.
                 string logDirectory = Environment.GetEnvironmentVariable(PerfLogDirEnvVar);
-                if (Directory.Exists(logDirectory))
+                Console.WriteLine("!!! DOTNET_PERFLOG_DIR: {0}", logDirectory);
+
+                if (!string.IsNullOrEmpty(logDirectory) && Directory.CreateDirectory(logDirectory).Exists)
                 {
                     eventListener = new PerformanceLogEventListener();
                     eventListener.Initialize(logDirectory);
+                    Console.WriteLine("!!! PerfLogEventListener will log to: {0}", logDirectory);
+                }
+                else
+                {
+                    Console.WriteLine("!!! PerfLogEventListener failed to initialize.");
+                    Console.WriteLine("!!! LogDirectory: {0}", logDirectory);
+                    Console.WriteLine("!!! DOTNET_PERFLOG_DIR: {0}", Environment.GetEnvironmentVariable(PerfLogDirEnvVar));
                 }
             }
-            catch
+            catch (Exception e)
             {
+                Console.WriteLine("!!! We threw!\nInner Exception: {0}\nMessage: {1}", e.InnerException, e.Message);
```

Comment:
Are Console.WriteLines the right way to log? If so, are they meant to be user-facing? If so, shouldn't we localize them? Either way, why so many exclamation points? Also, this should be much simpler if it's actually needed:
```suggestion
                Console.WriteLine(e.Message + "Inner Exception: " + e.InnerException);
```

Reply:
This was part of a debug commit that will be reverted when the PR is ready to go in.